02-OOP-Inheritance/Ex05/src/classEventEmitter.js

```diff
@@ -0,0 +1,32 @@
+class eventEmitter {
+	constructor(events) {
+		this.events (eventName, callbacks) = {
+			this.eventName: this.eventName,
+			this.callbacks: [],
+		};
+	}
+	on(eventName, callback) {
+		if(this.events.eventName === eventName) {
+			let i = this.events.callbacks.length;
+			this.events.callbacks[i+1] = callback;
+		}
+	}
+	emit(eventName) {
+		if (this.events.eventName === eventName) {
+			for (var i=0; i<=this.events.callbacks.length; i++) {
+				this.events.callbacks[i];
+			}
+		}
+	}
+	off(eventName, callback) {
```

Comment:
Just remove the pair of key: values for the event or set to undefined / null the array of events associated.

```

off (eventName) {
    let callbacks = this.events[eventName];

    if (callbacks) {
        callbacks = undefined;
    }
}
```